provider/provider_test.go

```diff
@@ -110,3 +211,112 @@ func maxDuration(dur time.Duration, t *testing.T, test func(t *testing.T)) {
 	case <-done:
 	}
 }
+
+// writeStackExport writes the stack export to the given path creating any directories needed.
+func writeStackExport(path string, snapshot *apitype.UntypedDeployment, overwrite bool) error {
+	if snapshot == nil {
+		return fmt.Errorf("stack export must not be nil")
+	}
+	dir := filepath.Dir(path)
+	err := os.MkdirAll(dir, 0755)
+	if err != nil {
+		return err
+	}
+	stackBytes, err := json.MarshalIndent(snapshot, "", "  ")
+	if err != nil {
+		return err
+	}
+	pathExists, err := exists(path)
+	if err != nil {
+		return err
+	}
+	if pathExists && !overwrite {
+		return fmt.Errorf("stack export already exists at %s", path)
+	}
+	return os.WriteFile(path, stackBytes, 0644)
+}
+
+// tryReadStackExport reads a stack export from the given file path.
+// If the file does not exist, returns nil, nil.
+func tryReadStackExport(path string) (*apitype.UntypedDeployment, error) {
+	stackBytes, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read stack export at %s: %v", path, err)
+	}
+	var stackExport apitype.UntypedDeployment
+	err = json.Unmarshal(stackBytes, &stackExport)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal stack export at %s: %v", path, err)
+	}
+	return &stackExport, nil
+}
+
+func exists(filePath string) (bool, error) {
+	_, err := os.Stat(filePath)
+	switch {
+	case err == nil:
+		return true, nil
+	case !os.IsNotExist(err):
+		return false, err
+	}
+	return false, nil
+}
+
+// cleanPlan removes fields that change every time, i.e. 'time' or 'seed'
+func cleanPlan(t *testing.T, plan map[string]interface{}) map[string]interface{} {
+	t.Helper()
+	if val, exists := plan["resourcePlans"]; exists {
+		resourcePlans := val.(map[string]interface{})
+		for _, v := range resourcePlans {
+			resourcePlan := v.(map[string]interface{})
+			delete(resourcePlan, "seed")
+
+			if goal, ok := resourcePlan["goal"]; ok {
+				g := goal.(map[string]interface{})
+				if g["type"] == "pulumi:providers:aws" {
+					if inputDiff, ok := g["inputDiff"]; ok {
+						idiff := inputDiff.(map[string]interface{})
+						if adds, ok := idiff["adds"]; ok {
+							a := adds.(map[string]interface{})
+							delete(a, "region")
```

Comment:
Why do we delete the region, that shouldn't change, right?

Reply:
It can change between what you set it to locally and what is set in CI. I ended up changing the test to set the region explicitly, but I thought to keep this in.

Reply:
But wouldn't change more in this case? E.g. `ARNs` or `URLs`

Reply:
Hmm that's true. I"ll remove it.